app/data/set.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Created by Jack on 10/06/2017.
+ */
+
+var fs = require('fs'),
+    crypto = require("crypto"),
+    local = require('../local.config.json'),
+    db = require('./database.js'),
+    path = require("path"),
+    async = require('async'),
+    backhelp = require("./backend_helpers.js");
+
+exports.version = "0.1.0";
+
+exports.createSet = function (data, callback) {
+    var finalSet;
+    var writeSucceeded = false;
+    async.waterfall([
+        // validate data.
+        function (cb) {
+            try {
+                backhelp.verify(data,
+                    [ "_id",
+                        "totalCards",
+                        "rarities",
+                        "image",
+                        "cards"
+                    ]);
+                if (!backhelp.validSetName(data._id))
+                    throw invalidSetName();
+            } catch (e) {
+                cb(e);
+                return;
+            }
+            cb(null, data);
+        },
+            // create the album in mongo.
+        function (setData, cb) {
+            var write = JSON.parse(JSON.stringify(setData));
+            write._id = setData.name + '_' + setData.rarity;
+            db.sets.insertOne(write, { w: 1, safe: true }, cb);
```

Comment:
Out of curiosity, what happens if the set already exists in the database and you try to insert it? May have to add error handling in that case